cl/beacon/handler/attestation_rewards.go

```diff
@@ -0,0 +1,450 @@
+			unslashedMatchingTargetBalanceIncrements += validator.EffectiveBalance()
+		}
+		if previousMatchingHeadAttester {
+			unslashedMatchingHeadBalanceIncrements += validator.EffectiveBalance()
+		}
+		return true
+	})
+	if err != nil {
+		return nil, err
+	}
+	// Then compute their total increment.
+	unslashedMatchingSourceBalanceIncrements /= beaconConfig.EffectiveBalanceIncrement
+	unslashedMatchingTargetBalanceIncrements /= beaconConfig.EffectiveBalanceIncrement
+	unslashedMatchingHeadBalanceIncrements /= beaconConfig.EffectiveBalanceIncrement
+	fn := func(index uint64, currentValidator solid.Validator) error {
+		baseReward, err := s.BaseReward(index)
+		if err != nil {
+			return err
+		}
+		var previousMatchingSourceAttester, previousMatchingTargetAttester, previousMatchingHeadAttester bool
+
+		if previousMatchingSourceAttester, err = s.ValidatorIsPreviousMatchingSourceAttester(int(index)); err != nil {
+			return err
+		}
+		if previousMatchingTargetAttester, err = s.ValidatorIsPreviousMatchingTargetAttester(int(index)); err != nil {
+			return err
+		}
+		if previousMatchingHeadAttester, err = s.ValidatorIsPreviousMatchingHeadAttester(int(index)); err != nil {
+			return err
+		}
+		totalReward := TotalReward{ValidatorIndex: int64(index)}
+		idealReward := IdealReward{EffectiveBalance: int64(currentValidator.EffectiveBalance())}
+
+		// check inclusion delay
+		if !currentValidator.Slashed() && previousMatchingSourceAttester {
+			var attestation *solid.PendingAttestation
+			if attestation, err = s.ValidatorMinPreviousInclusionDelayAttestation(int(index)); err != nil {
+				return err
+			}
+			proposerReward := (baseReward / beaconConfig.ProposerRewardQuotient)
+			maxAttesterReward := baseReward - proposerReward
+			idealReward.InclusionDelay = int64(maxAttesterReward / attestation.InclusionDelay())
+			totalReward.InclusionDelay = idealReward.InclusionDelay
+		}
+		// if it is not eligible for rewards, then do not continue further
+		if !(currentValidator.Active(prevEpoch) || (currentValidator.Slashed() && prevEpoch+1 < currentValidator.WithdrawableEpoch())) {
+			response.IdealRewards = append(response.IdealRewards, idealReward)
+			response.TotalRewards = append(response.TotalRewards, totalReward)
+			return nil
+		}
+		if inactivityLeak {
+			idealReward.Source = int64(baseReward)
+			idealReward.Target = int64(baseReward)
+			idealReward.Head = int64(baseReward)
+		} else {
+			idealReward.Source = int64(baseReward * unslashedMatchingSourceBalanceIncrements / rewardDenominator)
+			idealReward.Target = int64(baseReward * unslashedMatchingTargetBalanceIncrements / rewardDenominator)
+			idealReward.Head = int64(baseReward * unslashedMatchingHeadBalanceIncrements / rewardDenominator)
+		}
+		// we can use a multiplier to account for all attesting
+		var attested, missed uint64
+		if currentValidator.Slashed() {
+			attested, missed = 0, 3
+		} else {
+			if previousMatchingSourceAttester {
+				attested++
```

Comment:
@Giulio2002 
We could get rid of `attested`, which causing the error in your PR. Here is my suggestion, please take a look. 

```suggestion
		// we can use a multiplier to account for all attesting
                var missed uint64 = 3
		if !currentValidator.Slashed() {
	                if previousMatchingSourceAttester {
		               missed--
```